### PR TITLE
comply with new node representation

### PR DESCRIPTION
**Pull request by Karsten1987**
I think the `ros2lifecycle` is currently not working currently since https://github.com/ros2/ros2cli/pull/146 got merged.

### CODE REVIEW

**Comment by mjcarroll** [comment]
Yes, I was supposed to follow up on that, I didn't realize that #146 got merged.  Sorry for the inconvenience.

File: ros2lifecycle/ros2lifecycle/verb/set.py [hunk 1]
```diff
@@ -44,7 +44,12 @@ def main(self, *, args):  # noqa: D102
             node_names = get_node_names(
                 node=node, include_hidden_nodes=args.include_hidden_nodes)
 
-        if args.node_name not in node_names:
+        node_found = False
+        for node_name in node_names:
+            if args.node_name in node_name.full_name:
+                node_found = True
+                break
+        if not node_found:
```
Comment (dirk-thomas) [inline ros2lifecycle/ros2lifecycle/verb/set.py]: This could be much shorter: `if args.node_name not in {n.full_name for n in node_names}:`

Comment (Karsten1987) [reply ros2lifecycle/ros2lifecycle/verb/set.py]: of course there is a one-liner for python :) thanks!

File: ros2lifecycle/ros2lifecycle/api/__init__.py [hunk 2]
```diff
@@ -27,14 +27,14 @@ def get_node_names(*, node, include_hidden_nodes=False):
         node=node, include_hidden_nodes=include_hidden_nodes)
     service_names_and_types = get_service_names_and_types(node=node)
     return [
-        n for n in node_names
-        if _has_lifecycle(n, service_names_and_types)]
+        n.full_name for n in node_names
```
Comment (dirk-thomas) [inline ros2lifecycle/ros2lifecycle/api/__init__.py]: Would it be more consistent with the same function make in `ros2node` to return the "full" `n` here and let the callers pick the`full_name` from it?

Comment (Karsten1987) [reply ros2lifecycle/ros2lifecycle/api/__init__.py]: I can change it if you want. But given that this function anyway only really serves for the lifecycle implementation i didn't do it at first.

**Pull request merged**